extensions/wikia/AdEngine/js/video/vastUrlBuilder.js

```diff
@@ -47,31 +47,17 @@ define('ext.wikia.adEngine.video.vastUrlBuilder', [
 		return aspectRatio >= 1 || !isNumeric(aspectRatio) ? adSizes.horizontal : adSizes.vertical;
 	}
 
-	function buildAdUnit(slotParams, useMegaAdUnitBuilder) {
-		var adUnitBuilder = useMegaAdUnitBuilder ? megaAdUnitBuilder : regularAdUnitBuilder;
-		return adUnitBuilder.build(slotParams.pos, slotParams.src);
-	}
-
-	/**
-	 * Creates VAST URL from given parameters.
-	 * If `options.useMegaAdUnitBuilder` is not explicitly provided, the `context.opts.megaAdUnitBuilderEnabled` is used.
-	 * @param {number} aspectRatio
-	 * @param {object} slotParams
-	 * @param {object} options
-	 * @returns {string}
-	 */
 	function build(aspectRatio, slotParams, options) {
 		options = options || {};
 		slotParams = slotParams || {};
 
 		var correlator = options.correlator || Math.round(Math.random() * 10000000000),
-			useMegaAdUnitBuilder = options.useMegaAdUnitBuilder,
 			params,
 			url;
 
-		if (typeof useMegaAdUnitBuilder === 'undefined') {
-			useMegaAdUnitBuilder = adContext.getContext().opts.megaAdUnitBuilderEnabled;
-			log(['build', '`useMegaAdUnitBuilder` option is not defined. Fallback to `context.opts.megaAdUnitBuilderEnabled`.'], 'debug', logGroup);
+		if (!options.adUnit && adContext.get('opts.megaAdUnitBuilderEnabled')) {
+			// TODO: remove it after merge & release ADEN-5825 mobile-wiki, this is cache related backward compatibility
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

Reply:
Let's create a follow-up ticket for it